torch/_dynamo/variables/iter.py

```diff
@@ -263,10 +289,155 @@ def next_variable(self, tx):
             self.saved_index = (self.saved_index + 1) % len(self.saved)
             return self.item
         else:
-            # CPython here raises an exception. Since there is no python code, we have to manually setup the exception
-            # stack and raise the exception.
-            exception_vt = variables.BuiltinVariable(StopIteration).call_function(
-                self, [], {}
+            raise_observed_user_stop_iteration(self, tx)
+
+
+class ZipVariable(IteratorVariable):
+    """
+    Represents zip(*iterables)
+    """
+
+    _nonvar_fields = {
+        "index",
+        "strict",
+        *VariableTracker._nonvar_fields,
+    }
+
+    def __init__(
+        self,
+        iterables: List[Union[List[VariableTracker], VariableTracker]],
+        strict: bool = False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        assert isinstance(iterables, list)
+        # can be list[Variable] or VariableTracker (with next_variable implemented)
+        self.iterables = iterables
+        self.index = 0
+        self.strict = strict
+
+    def python_type(self):
+        return zip
+
+    def has_unpack_var_sequence(self, tx) -> bool:
+        return all(
+            isinstance(it, list) or it.has_unpack_var_sequence(tx)
```

Comment:
Why only list?

Reply:
`it` is either `VariableTracker` or `List[VariableTracker]` (from `unpack_var_sequence` called from the `Map/ZipVariable` callsite).